compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/CallGenerator.kt

```diff
@@ -202,20 +202,35 @@ class CallGenerator(statementGenerator: StatementGenerator) : StatementGenerator
         val irType = descriptor.type.toIrType()
 
         return if (getMethodDescriptor == null) {
-            call.callReceiver.call { dispatchReceiverValue, _ ->
-                val superQualifierSymbol = (call.superQualifier ?: descriptor.containingDeclaration as? ClassDescriptor)?.let {
-                    if (it is ScriptDescriptor) null // otherwise it creates a reference to script as class; TODO: check if correct
-                    else context.symbolTable.referenceClass(it)
+            call.callReceiver.call { dispatchReceiverValue, extensionReceiver ->
+                if (dispatchReceiverValue != null) {
+                    val superQualifierSymbol = (call.superQualifier ?: descriptor.containingDeclaration as? ClassDescriptor)?.let {
+                        if (it is ScriptDescriptor) null // otherwise it creates a reference to script as class; TODO: check if correct
+                        else context.symbolTable.referenceClass(it)
+                    }
+                    val fieldSymbol = context.symbolTable.referenceField(descriptor.resolveFakeOverride().original)
+                    IrGetFieldImpl(
+                        startOffset, endOffset,
+                        fieldSymbol,
+                        irType,
+                        dispatchReceiverValue.load(),
+                        IrStatementOrigin.GET_PROPERTY,
+                        superQualifierSymbol
+                    ).also { context.callToSubstitutedDescriptorMap[it] = descriptor }
```

Comment:
Probably doesn't matter right now, but this `also { ... }` call looks like it should apply to the IrGetFieldImpl below too, so maybe should be extracted out of the `if` statement

Reply:
Done.